src/main/scala/io/lenses/connect/secrets/providers/VaultSecretProvider.scala

```diff
@@ -162,4 +168,20 @@ class VaultSecretProvider() extends ConfigProvider with VaultHelper {
         )
     }
   }
+
+  private def readSecretFromVault(path: String): LogicalResponse = {
+    // Check if the path looks like it's referencing the vault database engine. If so, use the `database()` api of the
+    // vault client to obtain this secret.  Otherwise, fall back to the default key-vaule engine.
+    if (path.startsWith("database/creds/")) {
```

Comment:
We may want to have a test case for database creds. A good place to start from would probably be the [test case for kv secrets](https://github.com/lensesio/secret-provider/blob/master/src/test/scala/io/lenses/connect/secrets/providers/VaultSecretProviderTest.scala#L262).